src/visitable.h

```diff
@@ -58,6 +59,16 @@ class visitable
 
         /** Returns true if any item (including those within a container) matches the filter */
         bool has_item_with( const std::function<bool( const item & )> &filter ) const;
+
+        /**
+         * Removes items contained by this instance which match the filter
+         * @note if this instance itself is an item it will not be considered by the filter
+         * @param filter a UnaryPredicate which should return true if the item is to be removed
+         * @param count maximum number of items to remove or negative for unlimited. A count of zero is a no-op
+         * @return any items removed (items counted by charges are not guaranteed to be stacked)
+         */
+        std::list<item> remove_items_with( const std::function<bool( const item & )> &filter,
+                                           int count = -1 );
```

Comment:
Btw. the count parameter appears to be pretty much useless when removing items that are counted by charges. For example removing 10 *charges* of water. What do you supply as count? The player may have a stack of containers of 10 waters, or several contains of varying amounts of water.

Is there a plan to make this work with charges?